subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoTaskExtension.java

```diff
@@ -52,11 +54,12 @@ public String getAsArg() {
         }
     }
 
+    private Project project;
```

Comment:
> I'd assume that we still want to have some sort of method that let's us create a Provider from a Script for convenience reasons. Correct?

Do you mean keep `Project.provider(Callable)` or something similar for creating ad hoc `Provider` instances? If so, then yes.

> I think injecting a factory of a Provider into an extension is kind of tricky

It's actually quite easy with `DependencyInjectingInstantiator`.

There are other options too:

- Add a base class that extension classes can extend. This base class can provide some methods to do this.
- Inject directly into the field.
- Use abstract getters and setters.

I would possibly split this use case (can query the properties of an extension as a provider) as a separate story. Initially, if a task can accept a provider and you can create an ad hoc provider using something like `provider(Callable)`, then this is a good start.